File: hapi-fhir-structures-dstu3/src/test/java/ca/uhn/fhir/rest/client/ExtendedPatient.java
package ca.uhn.fhir.rest.client;

import org.hl7.fhir.dstu3.model.Patient;
import org.hl7.fhir.dstu3.model.StringType;

import ca.uhn.fhir.model.api.annotation.Child;
import ca.uhn.fhir.model.api.annotation.Description;
import ca.uhn.fhir.model.api.annotation.Extension;
import ca.uhn.fhir.model.api.annotation.ResourceDef;

@ResourceDef(name = "Patient", profile = ExtendedPatient.HTTP_FOO_PROFILES_PROFILE)
public class ExtendedPatient extends Patient {

	static final String HTTP_FOO_PROFILES_PROFILE = "http://foo/profiles/Profile";
  /**
	* 
	*/
	private static final long serialVersionUID = 1L;
	/**
	 * Each extension is defined in a field. Any valid HAPI Data Type
	 * can be used for the field type. Note that the [name=""] attribute
	 * in the @Child annotation needs to match the name for the bean accessor
	 * and mutator methods.
	 */
	@Child(name = "petName")
	@Extension(url = "http://example.com/dontuse#petname", definedLocally = false, isModifier = false)
	@Description(shortDefinition = "The name of the patient's favourite pet")
	private StringType myPetName;

	public StringType getPetName() {
		if (myPetName == null) {
			myPetName = new StringType();
		}
		return myPetName;
	}

	public void setPetName(StringType thePetName) {
		myPetName = thePetName;
	}

}

File: hapi-kotlin-test/src/main/kotlin/ca/uhn/fhir/rest/server/resources/ExtendedPatient.kt
package ca.uhn.fhir.rest.server.resources

import ca.uhn.fhir.model.api.annotation.Child
import ca.uhn.fhir.model.api.annotation.Extension
import ca.uhn.fhir.model.api.annotation.ResourceDef
import org.hl7.fhir.dstu3.model.CodeableConcept
import org.hl7.fhir.dstu3.model.Patient
import org.hl7.fhir.instance.model.api.IBaseResource

import java.util.Collections.emptyList

@ResourceDef(name = "Patient")
class ExtendedPatient : Patient() {

   @Child(name = "status")
   @Extension(
      url = "http://some.url",
      definedLocally = false,
      isModifier = false
   )
   var status: List<CodeableConcept> = emptyList()
}
